docs/developers/serializers.rst

```diff
@@ -0,0 +1,187 @@
+
+This allows you to make formatting or other changes before saving to disk, and
+conversely you can make changes to files before importing into Pootle.
+
+Serializers and deserializers are configured for a project. See the
+`django-admin`:"project_serializers" command for more information on configuring projects.
+
+To add your own serializers of deserializers you must add your own django application,
+and include it in your `INSTALLED_APPS` setting.
+
+This document explains what files you will need in your application to add serializers
+or deserializers.
+
+
+.. _serializers#custom-serializer:
+
+Adding a custom serializer
+--------------------------
+
+Lets add a `CustomSerializer` in a file called `serializers.py`.
+
+The serializer class will be called with 2 arguments, `store`, and
+`original_data`.
+
+The original data is a bare representation of the Store according to the `Project`'s
+`localfiletype`, unless the data has already been through another serialization filter
+already.
+
+The serializer class should implement a property called `output`, which should
+return the modified data. Out from a serializer class should be parseable
+by ttk.
+
+In this example, we strip whitespace from the original data, but only if the
+Store being serialized has an odd-numbered id.
+
+Serializers can make minor formatting changes, or they can change the structure
+of the document. If the content is changed in such a way that ttk will parse
+the result differently, then a matching deserializer will likely be required.
+
+.. code-block:: python
+
+   from pootle_store.store.serialize import StoreSerializer
+
+
+   class CustomSerializer(StoreSerializer):
+
+       @property
+       def output(self):
+           if self.store.id % 2:
+               return self.original_data.strip()
+           return self.original_data
+
+
+.. _serializers#custom-deserializer:
+
+Adding a custom deserializer
+--------------------------
+
+Lets add a `CustomDeserializer` in a file called `deserializers.py`.
+
+The deserializer class will be called with 2 arguments, `store`, and
+`original_data`.
+
+The deserializer class should implement a property called `output`, which should
+return the modified data. Output from a deserializer class should be parseable
+by ttk.
+
+In this example, we add whitespace to the original data, but only if the
+Store being serialized has an odd-numbered id.
+
+Deserializers can make minor formatting changes, or they can change the structure
+of the document. If the content is changed in such a way that ttk will parse
+the result differently, then a matching serializer will likely be required.
+
+.. code-block:: python
+
+   from pootle_store.store.serialize import StoreSerializer
+
+
+   class CustomSerializer(StoreSerializer):
+
+       @property
+       def output(self):
+           if self.store.id % 2:
+               return " %s " % self.original_data
+           return self.original_data
+
+
```

Comment:
@unho this is not the place for that

i am going to add docs for the `project_serializers` command, which will cover that